nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-file-authorizer/src/main/java/org/apache/nifi/authorization/FileAccessPolicyProvider.java

```diff
@@ -380,24 +413,55 @@ public void setNiFiProperties(NiFiProperties properties) {
 
     @Override
     public synchronized void inheritFingerprint(String fingerprint) throws AuthorizationAccessException {
-        parsePolicies(fingerprint).forEach(policy -> addAccessPolicy(policy));
+        final List<AccessPolicy> accessPolicies = parsePolicies(fingerprint);
+        inheritAccessPolicies(accessPolicies);
+    }
+
+    private synchronized void inheritAccessPolicies(final List<AccessPolicy> accessPolicies) {
+        addAccessPolicies(accessPolicies);
+    }
+
+    @Override
+    public synchronized void forciblyInheritFingerprint(final String fingerprint) throws AuthorizationAccessException {
+        final List<AccessPolicy> accessPolicies = parsePolicies(fingerprint);
+
+        if (isInheritable(accessPolicies)) {
+            logger.debug("Inheriting cluster's Access Policies");
+            inheritAccessPolicies(accessPolicies);
+        } else {
+            logger.info("Cannot directly inherit cluster's Access Policies. Will create backup of existing policies and replace with proposed policies");
+
+            try {
+                backupPolicies();
+            } catch (final JAXBException jaxb) {
+                throw new AuthorizationAccessException("Failed to backup existing policies so will not inherit any policies", jaxb);
+            }
+
+            purgePolicies(false);
+            addAccessPolicies(accessPolicies);
+        }
     }
 
     @Override
     public void checkInheritability(String proposedFingerprint) throws AuthorizationAccessException, UninheritableAuthorizationsException {
+        final List<AccessPolicy> accessPolicies;
         try {
             // ensure we can understand the proposed fingerprint
-            parsePolicies(proposedFingerprint);
+            accessPolicies = parsePolicies(proposedFingerprint);
         } catch (final AuthorizationAccessException e) {
             throw new UninheritableAuthorizationsException("Unable to parse the proposed fingerprint: " + e);
         }
 
         // ensure we are in a proper state to inherit the fingerprint
-        if (!getAccessPolicies().isEmpty()) {
+        if (!isInheritable(accessPolicies)) {
```

Comment:
1) `accessPolicies` are not used in `isInheritable`.
2) Does it make sense to still have this check if we support force inheriting? Failing on parse makes sense but not sure about this one.

Reply:
I agree. I don't think we should be checking that at all.

Reply:
Actually, after reviewing this a bit (it's been a while since I've visited this PR :)) it looks like this is necessary. In `AuthorizerCheck.checkInheritability`, this `UninheritableAuthorizationsException` gets caught and turned into a `FlowInheritability.notInheritable`.

This is important because we can only do the forced inheritance on startup. If the node is disconnected from the cluster and then re-connected while running, we cannot inherit the cluster's flow forcibly. So this `FlowInheritabiility` is returned back up the stack, and the `StandardFlowSynchronizer` to determine whether or not we should call the `forciblyInherit` based on the state of the instance.

Reply:
I see. Thanks for clarifying. I can confirm that this was preventing forceful inheritance while running.